HW0/P4/P4.py

```diff
@@ -65,12 +86,48 @@
 
     # B = ?
     # C = ?
+    A_matrix = propagation
+    B_diag = np.array([bx, by, bz, bvx, bvy, bvz])
+    B_matrix = np.diag(B_diag)
+    C_matrix = np.hstack([np.diag(np.array([rx, ry, rz])), np.zeros([3,3])])
 
     # Initial conditions for s0 and Sigma0
     # Compute the rest of sk using Eqs (2), (3), (4), and (5)
+    s2_matrix = np.zeros([6, K])
+    s2_matrix[:, 0] = np.array([0.0, 0.0, 2.0, 15.0, 3.5, 4.0]).transpose()
+    sig_diag = np.array([1, 1, 1, 1, 1, 1]) * 0.01
+    sig = np.diag(sig_diag)
+
+    def predictS():
```

Comment:
Your plot seemed a little odd since your filtered trajectory doesn't follow the original trajectory. I went through the entire code. All the functions are supposed to be matrix multiplications and not np.dot products, which is exactly what has caused the error in the plot.